### ISSUE TITLE
`Style/IfWithSemicolon` - undefined method `source' for nil:NilClass

### ISSUE THREAD

**Issue by snutij** [rubocop/rubocop#11996]
Hello 👋 I hope I'll give enough information, do not hesitate if you need more context.

Given a file `foo.rb`: 
```ruby
if 1; elsif 2; end

```

Which is a valid syntax: 
```
ruby -c foo.rb
Syntax OK
```

--------

## Expected behavior

Should not trigger an error, display the offense and propose a fix (reading the code I think this is the expected behavior, not sure to be honest, at least not trigger an error as it's a valid syntax).

## Actual behavior

`bin/rubocop -d --only Style/IfWithSemicolon foo.rb`
 
```
An error occurred while Style/IfWithSemicolon cop was inspecting foo.rb:1:0.
undefined method `source' for nil:NilClass
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/style/if_with_semicolon.rb:51:in `correct_elsif'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/style/if_with_semicolon.rb:39:in `autocorrect'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/style/if_with_semicolon.rb:32:in `block in on_normal_if_unless'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/base.rb:377:in `correct'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/base.rb:181:in `add_offense'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/style/if_with_semicolon.rb:31:in `on_normal_if_unless'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/mixin/on_normal_if_unless.rb:10:in `on_if'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.1/lib/rubocop/cop/commissioner.rb:107:in `public_send'
```

## Steps to reproduce the problem
In `IfWithSemicolon` the `on_normal_if_unless` receive a node: 

```ruby
(ruby) node
s(:if,
  s(:int, 1), nil,
  s(:if,
    s(:int, 2), nil, nil))
```

Then to propose a fix the cop calls the `correct_elsif`, as `if node.else_branch&.if_type?` is `true`

```ruby
(ruby) node.else_branch
s(:if,
  s(:int, 2), nil, nil)
(ruby)  node.else_branch&.if_type?
true
```
But it's interpolate`#{node.if_branch.source}` which is `nil`

```ruby
(ruby) node.condition.source
"1"
(ruby) node.if_branch
nil
```

https://github.com/rubocop/rubocop/blob/aaf1c3f6285e5e9e84d43c37944a7faaacb3bf60/lib/rubocop/cop/style/if_with_semicolon.rb#L23-L34

https://github.com/rubocop/rubocop/blob/aaf1c3f6285e5e9e84d43c37944a7faaacb3bf60/lib/rubocop/cop/style/if_with_semicolon.rb#L38-L45

https://github.com/rubocop/rubocop/blob/aaf1c3f6285e5e9e84d43c37944a7faaacb3bf60/lib/rubocop/cop/style/if_with_semicolon.rb#L47-L54

## RuboCop version

```
bundle exec rubocop -V
1.53.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [arm64-darwin22]
  - rubocop-minitest 0.31.0
  - rubocop-rake 0.6.0
  - rubocop-sorbet 0.7.0
```
